src/main/java/gift/repository/ProductRepository.java

```diff
@@ -9,34 +14,62 @@
 @Repository
 public class ProductRepository {
 
-    private final Map<Long, Product> products = new HashMap<>();
+    private final JdbcTemplate jdbcTemplate;
+
+    public ProductRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
     private final AtomicLong sequence = new AtomicLong(0);
 
+    private SimpleJdbcInsert insert;
+
+    @PostConstruct
+    public void init() {
+        this.insert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName("product")
+                .usingGeneratedKeyColumns("id");
+    }
+
     public Product save(String name, int price, String imageUrl) {
-        Long newId = sequence.incrementAndGet();
-        Product product = new Product(newId, name, price, imageUrl);
-        products.put(newId, product);
-        return product;
+        Map<String, Object> params = new HashMap<>();
+        params.put("name", name);
+        params.put("price", price);
+        params.put("image_url", imageUrl);
+        params.put("category_id", null);
```

Comment:
[컬렉션 팩토리](https://devbksheen.tistory.com/entry/%EB%AA%A8%EB%8D%98-%EC%9E%90%EB%B0%94-%EC%BB%AC%EB%A0%89%EC%85%98-%ED%8C%A9%ED%86%A0%EB%A6%ACCollection-Factory-%EC%86%8C%EA%B0%9C-%EB%B0%8F-%EC%82%AC%EC%9A%A9%EB%B2%95)를 통해 조금 더 깔끔하게 변경해볼 수 있을 것 같아요. 

```suggestion
        Map<String, Object> params = Map.of(
            "name", name,
            "price", price,
            "image_url", imageUrl,
            "category_id", null
        );
```